src/coreclr/nativeaot/System.Private.CoreLib/src/System/CrashInfo.cs

```diff
@@ -0,0 +1,400 @@
+                return false;
+
+            // Exception type names are not truncated because the full name is important to bucketing and usually not that long
+            if (!WriteStringValue("type"u8, exception.GetType().ToString()))
+                return false;
+
+            StackFrame[] stackFrames = new StackTrace(exception).GetFrames();
+            if (stackFrames.Length > 0)
+            {
+                if (!OpenValue("stack"u8, '['))
+                    return false;
+
+                int count = 0;
+                foreach (StackFrame frame in stackFrames)
+                {
+                    // Check if the stack frame limit has been hit
+                    if (++count > maxNumberStackFrames)
+                        break;
+
+                    if (!WriteStackFrame(frame, maxMethodNameSize))
+                        return false;
+                }
+
+                CloseValue(']');
+            }
+
+            AggregateException? aggregate = exception as AggregateException;
+            if (aggregate is not null || exception.InnerException is not null)
+            {
+                if (!OpenValue("inner"u8, '['))
+                    return false;
+
+                // Write as many inner exceptions that will fit
+                if (aggregate is not null)
+                {
+                    foreach (Exception ex in aggregate.InnerExceptions)
+                    {
+                        if (!WriteExceptionWithFallback(default, ex, maxMessageSize, maxNumberStackFrames, maxMethodNameSize))
+                            break;
+                    }
+                }
+                else
+                {
+                    WriteExceptionWithFallback(default, exception.InnerException, maxMessageSize, maxNumberStackFrames, maxMethodNameSize);
+                }
+
+                CloseValue(']');
+            }
+            CloseValue('}');
+            return true;
+        }
+
+        /// <summary>
+        /// Write an exception stack frame to the triage buffer
+        /// </summary>
+        /// <param name="frame">the stack frame instance to write</param>
+        /// <param name="maxNameSize">limits the size of the frame type name</param>
+        /// <returns>true - success, false - out of triage buffer space</returns>
+        private bool WriteStackFrame(StackFrame frame, int maxNameSize)
+        {
+            if (!OpenValue(default, '{'))
+                return false;
+
+            nint ip = frame.GetNativeIPAddress();
+            if (!WriteHexValue("ip"u8, (ulong)ip))
+                return false;
+
+            if (!WriteHexValue("offset"u8, frame.GetNativeOffset()))
+                return false;
+
+            string method = DeveloperExperience.GetMethodName(ip, out IntPtr _);
+            if (method != null)
+            {
+                if (!WriteStringValue("name"u8, method, maxNameSize, truncateLeft: true))
```

Comment:
If I am reading this correctly, the method name includes signature. It means that truncating from left is going to leave us with tail of the signature. I do not think that is very useful.

Is there any prior art , experience or data analysis with truncating method names from left? I think we should stick with regular truncating from right. It is what one would expect and it seems to work at least as good, or even better, than truncating from right.

Reply:
I was thinking this from the perspective of AsyncStateMachine and boxes, where names are long and you end up having useful info only in between the `<>` at the end. But if this signature includes all the parameters, that's more likely to be an issue in the end. For example:

`System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1[[System.Threading.Tasks.VoidTaskResult, System.Private.CoreLib],[MyFooNamespace.MyFooProgram+<DeepRecursion>d__1, aggregate]]` The only part that's useful from a diagnostic standpoint is `DeepRecursion` which is past the 100 mark.

Reply:
In this specific example, you should also see the DeepRecursion method frame itself, so the diagnostic information about the DeepRecursion method itself should still be there.

Some tools hide the async infrastructure frames as non-interesting. If we trim the name from left, it won't be really possible to detect and hide the async infrastructure frames.

Reply:
I'm going to remove the truncateLeft: true parameter at this call, but leave the logic in WriteStringValue for now.  I can go back and clean it up once we figure out the best overall fallback policy (not just how the strings are truncate, but also the other limits).

Reply:
I was using a test that does async blocking to it gets unscheduled and the frame is gone. Thanks, that makes sense.